pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -120,6 +123,45 @@ func (s *SecretsManager) UpdateData(ctx context.Context, crt *cmapi.Certificate,
 	return err
 }
 
+func updateSecretWithAdditionalOutputFormats(crt *cmapi.Certificate, secret *corev1.Secret, data SecretData) error {
+	if crt.Spec.AdditionalOutputFormats == nil {
+		delete(secret.Data, cmapi.AdditionalKeyOutputFormatDERKey)
+		delete(secret.Data, cmapi.AdditionalOutputFormatPEMKey)
+		return nil
+	}
+
+	additionalOutputFormatDER := false
+	additionalOutputFormatPEM := false
+
+	for _, f := range crt.Spec.AdditionalOutputFormats {
+		switch f.Type {
+		case cmapi.AdditionalKeyOutputFormatDER:
+			additionalOutputFormatDER = true
+		case cmapi.AdditionalOutputFormatCombinedPEM:
+			additionalOutputFormatPEM = true
+		default:
+			return fmt.Errorf("Unknown additional output format %s", f.Type)
+		}
+	}
+
+	if additionalOutputFormatDER {
+		// Store binary format of the private key
+		block, _ := pem.Decode(data.PrivateKey)
+		secret.Data[cmapi.AdditionalKeyOutputFormatDERKey] = block.Bytes
+	} else {
+		delete(secret.Data, cmapi.AdditionalKeyOutputFormatDERKey)
+	}
+
+	if additionalOutputFormatPEM {
+		// Combine tls.key and tls.crt
+		secret.Data[cmapi.AdditionalOutputFormatPEMKey] = []byte(strings.Join([]string{string(data.PrivateKey), string(data.Certificate)}, "\n"))
+	} else {
+		delete(secret.Data, cmapi.AdditionalOutputFormatPEMKey)
+	}
+
+	return nil
+}
```

Comment:
This is fine as is, but heads up that this will look a bit different after https://github.com/jetstack/cert-manager/pull/4638